### ISSUE TITLE
Duplicate locals are emitted for switch expression arms

### ISSUE THREAD

**Issue by davidwengier** [dotnet/roslyn#50301]
See this comment: https://github.com/dotnet/roslyn/pull/50270/files#r552284955

This issue is referenced in source.

For a simple switch expression:

```cs
class C
{
    static string M(object o)
    {
        return o switch
        {
            int i => $"Number: {i}",
            _ => "Don't know"
        };
    }
}
```

There are nested scoped emitted, each with locals that have the same name, for `i`:

```xml
      <scope startOffset="0x0" endOffset="0x3d">
        <scope startOffset="0x1" endOffset="0x3b">
          <local name="i" il_index="0" il_start="0x1" il_end="0x3b" attributes="0" />
          <scope startOffset="0x16" endOffset="0x2b">
            <local name="i" il_index="0" il_start="0x16" il_end="0x2b" attributes="0" />
          </scope>
        </scope>
      </scope>
```

This also shows up during debugging (note the duplicate `i`s):
![image](https://user-images.githubusercontent.com/754264/103845115-7bd69b80-50ef-11eb-8a38-9aebdc634267.png)

/cc @tmat @jcouv

**Comment by jcouv** [comment]
I think the problem comes from `LowerSwitchExpression` (snippet below). It results in lowered nodes with the local listed as declared in a containing spill sequence while it's also declared by a switch arm in the spill sequence's result.

```csharp
                        // Lifetime of these locals is expanded to the entire switch body, as it is possible to
                        // share them as temps in the decision dag.
                        outerVariables.AddRange(arm.Locals);

                        // Note the language scope of the locals, even though they are included for the purposes of
                        // lifetime analysis in the enclosing scope.
                        result.Add(new BoundScope(arm.Syntax, arm.Locals, statements));
 ```

**Comment by jcouv** [comment]
Fix was merged fro 16.10p1
